src/runtime_src/core/pcie/tools/xbmgmt/xmc.cpp

```diff
@@ -119,6 +119,11 @@ int XMC_Flasher::xclUpgradeFirmware(std::istream& tiTxtStream) {
         return -EINVAL;
     }
 
+    if (fixedSC()) {
+        std::cout << "INFO: fixed SC version, skip upgrading" << std::endl;
+        return -EINVAL;
+    }
+
```

Comment:
if we do the check here, xbmgmt flash --sc_firmware will also bail out? i think we want to flash unconditionally when people use this debug command.

Reply:
Yes, I released this restriction. We should allow user flash it via this debugging interface. Given, the U30 project might not have a full shell which contains MCS in its xsabin at least for this release, and they might use sc_firmware to flash the firmware. To avoid massive flashing issues, I just add one more checking and only when user answered y(Yes), we allow them to do it what's so ever. Examples below:

`
root@xsjyliu50:/scratch/davidz# ./xbmgmt flash --sc_firmware --path U30_SC-FW-OEM_6_3_1.txt --card 0000:da:00.1
INFO: SC version is FIXED, should not be flashed!
Are you sure you wish to proceed? [y/n]: n
Action canceled.

root@xsjyliu50:/scratch/davidz# ./xbmgmt flash --sc_firmware --path U30_SC-FW-OEM_6_3_1.txt --card 0000:da:00.1
INFO: SC version is FIXED, should not be flashed!
Are you sure you wish to proceed? [y/n]: y
INFO: found 6 sections
...............................
`

Reply:
Let's remove this check when we start final testing to make sure CMC can return error on slave mgmt pf when we try to flash SC firmware. We do not want to add any workaround here for U30 in this regard.